### DISCUSSION TITLE
Relative paths

### DISCUSSION THREAD

**Discussion by daviddeacon** [Autodesk/maya-usd#1581] [Q&A]
Hi,

I'm working with Maya 2022, Version 0.10.0 on macOS, all of my USD paths are being written as absolute paths. This includes SubLayers and References. I can't find an option to write these as relative paths. This is a major roadblock to us rolling out USD into the studio pipeline as we need relative paths for a few workflows. eg, project files stored locally and on a network drive, and our cross platform render farm.

Also the main stage path is saved as an absolute path, this would be great if it could be stored relative to the Maya project.

I've been saving as usda files and can manually edit them to be relative and they do work. Even using `..` to go up a folder when a usd file isn't directly next to the file that is using it.

Is this expected behaviour or something I'm missing?

David.

**Comment by daviddeacon** [comment]
I just wanted to check in on this. While looking for workaround/solutions to not having relative paths. I came across this thread that discusses why relative paths are on by default in Solaris. 

https://www.sidefx.com/forum/topic/76401/

If there is no current way to save all USD files with relative paths I will create an issue from this discussion.

Thanks,
David.

**Comment by wtelford** [comment]
Hey David,
First Maya USD will preserve layers already authored as relative paths. 

Second, with regards to authoring with relative paths, right now they can be created for layers in two ways. 
1. With the "Load Sublayer" dialogue in the Layer Editor. 
![image](https://user-images.githubusercontent.com/1900693/127570882-13581f46-4658-4d59-98ff-a87c1673e4ea.png)
2. By passing a relative path to the "Save Stage" dialogue in the layer editor.

Unfortunately looking in to this it appears there may be a bug with method two that we are now investigating.

Our current reference support is very basic and we are working on designs for a more advanced reference/payload workflow.
I expect relative paths to be a requirement for that design.

**Reply by santosd** [reply]
The bug in regards to method 2 is now resolved MAYA-113008 so it should no longer be an issue.
